oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java

```diff
@@ -76,6 +76,15 @@ public class FilterProviderImpl implements FilterProvider {
     private final Map<String, String> validatedPrincipalNamesPathMap = Maps.newConcurrentMap();
     private final Map<String, String> unsupportedPrincipalNames = Maps.newConcurrentMap();
 
+    /**
+     * Constructor to use outside OSGi containers
+     * @param oakPath the repository path where the principals are located
+     * @since 1.54
+     */
+    public FilterProviderImpl(String oakPath) {
```

Comment:
@kwin , please add @NotNull annotation for the path param

Reply:
Done